src/Resolution.ts

```diff
@@ -44,68 +54,127 @@ export default class Resolution {
   constructor({
     blockchain = true,
     api = DefaultAPI,
-  }: { blockchain?: Blockchain; api?: API } = {}) {
+  }: { blockchain?: Blockchain | boolean; api?: API } = {}) {
     this.blockchain = !!blockchain;
     if (blockchain) {
-      if (blockchain == true) {
+      if (blockchain === true) {
         blockchain = {};
       }
-      if (blockchain.ens === undefined) {
-        blockchain.ens = true;
+      const web3provider = blockchain.web3Provider;
+      if (web3provider) {
+        console.warn('Usage of `web3Provider` option is deprecated. Use `provider` option instead for each individual blockchain');
       }
-      if (blockchain.zns === undefined) {
-        blockchain.zns = true;
-      }
-      if (blockchain.cns === undefined) {
-        blockchain.cns = true;
-      }
-      if (blockchain.ens) {
-        this.ens = new Ens(
-          blockchain.ens,
-          blockchain.web3Provider as Web3Provider,
-        );
+      const ens = this.normalizeSource(blockchain.ens, web3provider);
+      const zns = this.normalizeSource(blockchain.zns);
+      const cns = this.normalizeSource(blockchain.cns, web3provider);
+
+      if (ens) {
+        this.ens = new Ens(ens);
       }
-      if (blockchain.zns) {
-        this.zns = new Zns(blockchain.zns);
+      if (zns) {
+        this.zns = new Zns(zns);
       }
-      if (blockchain.cns) {
-        this.cns = new Cns(
-          blockchain.cns,
-          blockchain.web3Provider as Web3Provider,
-        );
+      if (cns) {
+        this.cns = new Cns(cns);
       }
     } else {
-      this.api = new Udapi(api.url);
+      this.api = new Udapi(api);
     }
   }
 
   /**
    * Creates a resolution with configured infura id for ens and cns
    * @param infura infura project id
+   * @param network ethereum network name
    */
-  static infura(infura: string): Resolution {
+  static infura(infura: string, network: string = 'mainnet'): Resolution {
     return new this({
       blockchain: {
-        ens: { url: signedInfuraLink(infura), network: 'mainnet' },
-        cns: { url: signedInfuraLink(infura), network: 'mainnet' },
+        ens: { url: signedInfuraLink(infura, network), network },
+        cns: { url: signedInfuraLink(infura, network), network },
       },
     });
   }
 
   /**
    * Creates a resolution instance with configured provider
-   * @param provider - any provider with sendAsync function impelmented
+   * @param provider - any provider compatible with EIP-1193
+   * @see https://eips.ethereum.org/EIPS/eip-1193
    */
-  static provider(provider: Web3Provider): Resolution {
-    return new this({ blockchain: { web3Provider: provider } });
+  static fromEipEthProvider(provider: Provider): Resolution {
```

Comment:
```suggestion
  static fromEip1193Provider(provider: Provider): Resolution {
```
Currently it stands for `from Ethereum Improvement Proposal Eth Provider`, which imo is tautology and isn't specific enough (what if there appears another proposal for web3 interface?)